### PR TITLE
WID-247 ConnectivityMatrixEditor widget

### PR DESCRIPTION
**Pull request by peeplika**

### CODE REVIEW

File: tvbwidgets/ui/connectivity_matrix_editor_widget.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
+import math
```
Comment (915-Misan-Teodora) [inline tvbwidgets/ui/connectivity_matrix_editor_widget.py:1]: You should include the header file as you did for SpaceTime widget file.

File: tvbwidgets/ui/connectivity_matrix_editor_widget.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    def _make_header(self):
+        self.header = widgets.HBox()
+        options = ["Quadrant 1", "Quadrant 2", "Quadrant 3", "Quadrant 4"]
+        self.quadrants = widgets.Dropdown(options = options)
+        self.quadrants.observe(self._on_quadrant_select, names = ["value"])
+        self.cell_value = widgets.Text(description = "value")
+        self.button = widgets.Button(description = "Change")
+        self.button.on_click(lambda change :self.on_apply_change(change))
+        self.cell_value.layout.visibility = "hidden"
+        self.button.layout.visibility = "hidden"
```
Comment (915-Misan-Teodora) [inline tvbwidgets/ui/connectivity_matrix_editor_widget.py:34]: When you create a new widget you can add its layout directly in its definition. This approach is cleaner than adding it afterwards.
ex: self.button = widgets.Button(description = "Change", layout=widgets.Layout(width='80px', visibility='hidden'))

**Comment by 915-Misan-Teodora** [comment]
I tested the widget and I have some suggestions:

- the canvas is really big on my screen - I saw that width/height are hardcoded, maybe you can let the user choose its dimensions
- when you click on a cell the page is scrolling up and it selects another cell than the desired one(probably bcs of its size- I tested with a smaller size and it does not scroll the page anymore) - maybe there is a way to disable the scroll or just make the canvas smaller
- the values from the colored legend matrix could be displayed from the upper bound value to the lower bound (as in tvb-web)
- when changing a cell value to a bigger one not from the range(to say 30 or 300) the others cells color should change as well because of the scale - you can test it in tvb-web to see the differences
- the labels of the region are not so visible

**Comment by romina1601** [comment]
Hi @peeplika ,
I tested the widget and there is something not working as expected.

I tried using other connectivities, not the default one. For this I added an argument in the `from_file()` method like this: `conn = Connectivity().from_file("connectivity_66.zip")` or `conn = Connectivity().from_file("connectivity_96.zip")`. In these cases, the labels are not aligned with the connectivity cells and/or the connectivity cells are overlapping:

- connectivity_66.zip:
image
- connectivity_96.zip:
image

As you can imagine, modifying the cell values doesn't work either, so could you take a look at this?

**Comment by romina1601** [comment]
hey @peeplika ,
thanks for the changes! works and looks good, but I have a small observation. I saw that you made the window scrollable, but could you make the height of this window a bit bigger?
image

Also, I saw that after clicking on the save button, the ID displayed in the dropdown list does get updated. Could you change it such that the ID of the newly saved connectivity is displayed?

**Review by romina1601 — APPROVED** [review]

**Pull request merged**